src/platform/platform_posix.c

```diff
@@ -337,101 +337,83 @@ CxPlatEventInitialize(
     _In_ BOOLEAN InitialState
     )
 {
-    CXPLAT_EVENT_OBJECT* EventObj = NULL;
     pthread_condattr_t Attr = {0};
     int Result;
 
-    EventObj = CXPLAT_ALLOC_NONPAGED(sizeof(CXPLAT_EVENT_OBJECT), QUIC_POOL_EVENT);
+    Event->AutoReset = !ManualReset;
```

Comment:
Any reason not to inline all of this in the header (function or macro, I don't care which)?

Reply:
Would we inline all 5 functions, or just the 3 simple ones? Theres no reason not to inline these, especially without the allocation anymore.

Reply:
Everything